include/ozo/impl/result.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ozo/error.h>
+#include <ozo/type_traits.h>
+#include <ozo/concept.h>
+#include <ozo/impl/row.h>
+
+namespace ozo::impl {
+
+template <typename Rows, typename OidMap, typename Iterator>
+Require<Iterable<Rows> && (ForwardIterator<Iterator> || InsertIterator<Iterator>), error_code>
+convert_result(Rows&& rows, const OidMap& oid_map, Iterator result) {
+    error_code ec = error::ok;
+    for (auto row : rows) {
+        if constexpr (ForwardIterator<Iterator>) {
+            ec = convert_row(std::move(row), oid_map, *result++);
+        } else if constexpr (InsertIterator<Iterator>) {
+            using row_value_t = typename Iterator::container_type::value_type;
+
+            auto result_row = row_value_t{};
+            ec = convert_row(std::move(row), oid_map, result_row);
+            if (!ec) {
+                result = std::move(result_row);
+            }
+        }
+        if (ec) {
+            return ec;
+        }
+    }
+    return ec;
+}
+
+template <typename Rows>
+class result {
+    using rows_t = std::decay_t<Rows>;
+public:
+    result(rows_t&& rows) : rows_(std::move(rows)) {}
+
+    template <typename Iterator, typename OidMap>
+    error_code convert(Iterator&& result, const OidMap& oid_map) {
```

Comment:
IMO ```in``` or something like this is better than name ```result``` which is also a name of the class.